src/System/Wlog/PureLogging.hs

```diff
@@ -115,17 +116,50 @@ runNamedPureLog
 runNamedPureLog (NamedPureLogger action) =
     askLoggerName >>= (`usingLoggerName` runPureLog action)
 
--- | Similar to 'launchPureLog', but provides logger name from current context.
+{- | Similar to 'launchPureLog', but provides logger name from current context.
+
+Running the 'NamedPureLogger' gives us the pair  of target and the list of 'LogEvent's,
+wrapped in 'Monad' from where using the fact that @(,)@ is 'Functor' logging can be triggered.
+
+==== __Example__
+
+@
+  newtype PureSmth a = ...
+      deriving (MonadSmth, ...)
+
+  evalPureSmth :: PureSmth a -> a
+
+  makeField    :: MonadSmth m => Data -> m Field
+
+  run :: (MonadIO m, WithLogger m) => m ()
+  run = do
+      data  <- getData
+      -- field :: Field
+      field <- launchNamedPureLog (pure . evalPureSmth)
+                                  (makeField data)
+      --       ^ logging happens here
+      ...
+@
+
+-}
 launchNamedPureLog
     :: (WithLogger n, Monad m)
     => (forall f. Functor f => m (f a) -> n (f b))
     -> NamedPureLogger m a
     -> n b
-launchNamedPureLog hoist' (NamedPureLogger action) = do
+launchNamedPureLog hoist' namedPureLogger = do
     name <- askLoggerName
-    (logs, res) <- hoist' $ swap <$> usingLoggerName name (runPureLog action)
+    (logs, res) <- hoist' $ swap <$> usingNamedPureLogger name namedPureLogger
     res <$ dispatchEvents logs
 
+-- | Similar to 'launchNamedPureLog', but lift the value of function argument.
+launchNamedPureLogWithPure
```

Comment:
Yeah, this name is not really good... :)
But I can't think of really better name. Maybe `Pure` part is redundant and we can call it just `launchNamedPureLogWith`.